fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanScheduleService.java

```diff
@@ -54,7 +54,7 @@ public void regenerateRepaymentSchedule(final Loan loan, final ScheduleGenerator
     }
 
     public void recalculateScheduleFromLastTransaction(final Loan loan, final ScheduleGeneratorDTO generatorDTO) {
-        if (loan.repaymentScheduleDetail().isInterestRecalculationEnabled()) {
+        if (loan.repaymentScheduleDetail().isInterestRecalculationEnabled() && !loan.isChargedOff()) {
```

Comment:
When you call loan.isChargedOff(), it simply checks whether the loan has been charged off. As a result, you might skip the recalculation for installments that have dates earlier than the charge-off transaction. As far as I understand, you need to stop recalculating interest only AFTER the date of the charge-off transaction. Perhaps my PR could help you: https://github.com/apache/fineract/pull/4236